### DISCUSSION TITLE
[SOLVED] Sonoff: enable 'vcc' in mqtt state messages via console

### DISCUSSION THREAD

**Discussion by RaSTuS26** [arendst/Tasmota#10744] [Q&A] [answered]
I've noticed vcc is now missing in mqtt tele/state messages. Is there any way to enable this behaviour from the console?

There's the "#define USE_ADC_VCC" in my_user_config.h, but I don't want to really go through the trouble of flashing them all (approx 20) again manually.

**Comment by barbudor** [comment, marked as answer]
Hi
This is a build only option. As of today you can't change it without using a binary that has this option.
From [BUILDs.md](https://github.com/arendst/Tasmota/blob/development/BUILDS.md), this is by not enabled by default except for `tasmota-lite.bin` and `tasmota-ir.bin` (and `tasmota-minimal` but I don't count it as a normal binary)

So if you need `tasmota.bin` or `tasmota-sensors.bin`, you need to recompile and upgrade.
Alternatively, you can connect A0 to VCC with a voltage divider but I would expect hardware change to be more annoying than recompiling.

IMHO, this feature is more or less useless as you are measuring the VCC which is after the voltage regulator so supposed to be always 3.3V. For battery operated device, it is much more useful to get the battery voltage before the regulator using a voltage divider.
The only case where I would see that usefull is when oering the ESP with a LiPoFe4 battery without regulator.

**Reply by barbudor** [reply]
Don't forget that enabling ` USE_ADC_VCC` prevent totally the usage of the analog input for other uses.
(This is an ESP8266 limitation, not related to Tasmota nor Arduino nor Espressif SDK)

**Reply by RaSTuS26** [reply]
Thanks @barbudor, not really what I was hoping to hear, just spent 2 days reflashing all my sonoffs, now I'll need to repeat it all to get vcc status back, ugh.

I'll never need analog input, and my OpenHAB setup uses it, to disable it there will make all the pages line output uneven, but that'd probably be easier, I can look for something else to populate that space with I suppose.

EDIT: I'll just go through the mqrt STATE and SENSOR payloads and see if I can spot a likely candidate.

**Reply by sfromis** [reply]
What benefit do you see in getting that pretty useless VCC value?

**Reply by RaSTuS26** [reply]
As I mentioned it helps to balance the page in OpenHAB so there is not an orphan line at the bottom of the section.  I'm now using ENERGY.Factor instead, just as useless to me, but I do like the sections in OH to look balanced.

**Comment by RaSTuS26** [comment]
![OpenHAB_POWpage](https://user-images.githubusercontent.com/55551230/106266303-fcfafb80-6273-11eb-9720-2eedb229b6e5.png)

**Reply by meingraham** [reply]
An alternative to just displaying a low value metric just to align/balance your sitemap is to define an "empty" text Item and place it anywhere you want to "fill" a gap so that the next item to display ends up in the position you want it.

**Reply by RaSTuS26** [reply]
Thanks for the suggestion, that doesn't solve anything really, it still looks unbalanced and ugly, I've used that trick many times before and just don't like it.  I'm not worried about next items being bought up, each device is in it's own sitemap frame anyway so no chance of them being muddled together.

**Comment by RaSTuS26** [comment]
@barbudor It would be nice to add USE_ADC_VCC to the 'option' list, much easier to be able to toggle it rather than doing a full build and flash routine.  For a start it means taking the device apart to wire it up to a serial TTL adapter before you even consider the building and flashing.  A laborious approach to simply change a single option, especially when many devices are involved.

**Reply by ascillato** [reply]
That is not possible. Use vcc is a hardware option that is a bit enabled in the firmware. This can not be changed AFTER flashing. Is how the ESP8266 chip works. If you want the useless VCC you need to recompile and do an upgrade by OTA.
At runtime there isn't any firmware that can change the VCC option.

**Comment by RaSTuS26** [comment]
Oh well, that's it I suppose, I'm definitely not going through all that again for 20 odd devices, I'll just stick with ENERGY.Factor instead.

I'll close this in 24hrs, just in case any further responses come in, global timezones and all that.

**Reply by ascillato2** [reply]
Link to the old issue where this was already explained and discussed:

https://github.com/arendst/Tasmota/issues/5671

**Reply by RaSTuS26** [reply]
Thanks for the link, I did go through 7 or 8 pages of issues (and many google searches) looking for something related before starting this discussion, but missed that particular issue.  Changelogs going way back too.
